astropy/nddata/nduncertainty.py

```diff
@@ -484,6 +565,142 @@ class _VariancePropagationMixin:
+                    and to_variance(self.unit) != self.parent_nddata.unit**2
+                ):
+                    # If the uncertainty has a different unit than the result we
+                    # need to convert it to the results unit.
+                    this = to_variance(self.array << self.unit).to(result_unit_sq).value
+                else:
+                    this = to_variance(self.array)
+
+                return from_variance(this, axis=axis)
+
+    @staticmethod
+    def _get_err_at_extremum(measurement, uncertainty, extremum):
+        """
+        Return the value of the ``uncertainty`` array at the indices
+        which satisfy the ``extremum`` function applied to the ``measurement`` array,
+        where we expect ``extremum`` to be np.argmax or np.argmin, and
+        we expect a two-dimensional output.
+
+        Assumes the ``measurement`` and ``uncertainty`` array dimensions
+        are ordered such that the zeroth dimension is the one to preserve.
+        For example, if you start with array with shape (a, b, c), this
+        function applies the ``extremum`` function to the last two dimensions,
+        with shapes b and c.
+
+        This operation is difficult to cast in a vectorized way. Here
+        we implement it with a list comprehension, which is likely not the
+        most performant solution.
+        """
+        return np.array(
+            [
+                err[np.unravel_index(extremum(meas), meas.shape)]
+                for meas, err in zip(measurement, uncertainty)
+            ]
+        )
+
+    def _get_err_at_min(self, axis):
+
+        if axis is not None and not hasattr(axis, "__len__"):
+            # this is a single axis:
+            axis = [axis]
+
+        all_axes = np.arange(self.array.ndim)
+
+        if axis is None:
+            # collapse over all dimensions
+            ind = np.ma.argmin(np.asanyarray(self.parent_nddata).ravel())
+            return self.array.ravel()[ind]
+
+        else:
+            # collapse an ND array over arbitrary dimensions:
+            preserve_axes = [ax for ax in all_axes if ax not in axis]
+            meas = np.ma.masked_array(
+                _move_preserved_axes_first(self.parent_nddata.data, preserve_axes),
+                _move_preserved_axes_first(self.parent_nddata.mask, preserve_axes),
+            )
+            err = _move_preserved_axes_first(self.array, preserve_axes)
+            return _unravel_preserved_axes(
+                self.parent_nddata.data,
+                self._get_err_at_extremum(meas, err, np.ma.argmin),
+                preserve_axes,
+            )
+
+    def _get_err_at_max(self, axis):
+
+        if axis is not None and not hasattr(axis, "__len__"):
+            # this is a single axis:
+            axis = [axis]
+
+        all_axes = np.arange(self.array.ndim)
+
+        if axis is None:
+            # collapse over all dimensions
+            ind = np.ma.argmax(np.asanyarray(self.parent_nddata).ravel())
```

Comment:
I think this is the only line that is different between `_get_err_at_max` and `_get_err_at_min` -- can these two methods be combined if you add another argument?

Reply:
I think so! This was my original intention, but there _were_ subtleties that prevented me from doing this in earlier commits. I think those few divergences are all gone as of the last few commits, so I'll do that now.